src/Microsoft.AspNetCore.Mvc.Razor.Precompilation.Design/Internal/ViewCompilationInfo.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Mvc.Razor.Compilation;
+using Microsoft.AspNetCore.Razor.CodeGenerators;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Precompilation.Design.Internal
+{
+    public class ViewCompilationInfo
```

Comment:
This type is being used for two different purposes. 1 parsing a view and then determining errors at parse time. 2 expecting the GeneratorResults to be compiled and TypeName extracted so that this can be passed to the `ViewCollectionGenerator` (currently `CodeGenerator`).